src/components/admin/SearchableSelector.jsx

```diff
@@ -0,0 +1,125 @@
+/* eslint react/jsx-no-bind: 0 */
+
+import React from 'react';
+import update from 'react-addons-update';
+import _ from 'lodash';
+
+import BaseComponent from 'lib/BaseComponent';
+import SearchBar from 'components/admin/SearchBar';
+
+
+// material-ui
+import Chip from 'material-ui/Chip';
+
+class ObjectChip extends BaseComponent {
+  constructor(props) {
+    super(props);
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onClick() {
+    this.props.onDelete(this.props.id);
+  }
+
+  render() {
+    return (
+      <Chip
+        onRequestDelete={this.onClick}
+        style={this.props.style}
+      >
+        {this.props.children}
+      </Chip>
+    );
+  }
+}
+
+export default class EditObjectsForm extends BaseComponent {
```

Comment:
We can slowly start going away from our weird components, and you can just extend `React.PureComponent` here